pkgs/development/libraries/libfprint-2-tod1-elan/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, lib, fetchzip, fetchgit, libfprint-tod, openssl, unzip }:
+
+stdenv.mkDerivation {
+  pname = "libfprint-2-tod1-elan";
+  version = "0.0.8";
+
+
+  src = fetchzip {
+    url = "https://download.lenovo.com/pccbbs/mobiles/r1slf01w.zip";
+    sha256 = "sha256-eY3QsoiBMVuIlT9XY2wI5kc89opgHE7nQa9n71vFpsM=";
+  };
+
+  nativeBuildInputs = [ unzip ];
```

Comment:
Just making sure, have you tested this package? This seems like something we would need to [patchelf](https://nixos.wiki/wiki/Packaging/Binaries)

Reply:
Yes, the driver is getting loaded and is working fine with  libfprint-tod.
```
❯ sudo systemctl status fprintd.service          
Place your finger on the fingerprint reader
● fprintd.service - Fingerprint Authentication Daemon
     Loaded: loaded (/etc/systemd/system/fprintd.service; linked; preset: enabled)
    Drop-In: /nix/store/7vb6kgw1cyja6j7vyiraxv97qxbdfjjg-system-units/fprintd.service.d
             └─overrides.conf
     Active: active (running) since Wed 2023-11-01 05:18:03 CET; 2s ago
       Docs: man:fprintd(1)
   Main PID: 141456 (fprintd)
         IP: 0B in, 0B out
         IO: 0B read, 0B written
      Tasks: 7 (limit: 37922)
     Memory: 3.3M
        CPU: 80ms
     CGroup: /system.slice/fprintd.service
             └─141456 /nix/store/8md2307rwb0cp6zghcs329387hj2dw7y-fprintd-tod-1.90.9/libexec/fprintd

Nov 01 05:18:03 thinkbook14 systemd[1]: Starting Fingerprint Authentication Daemon...
Nov 01 05:18:03 thinkbook14 systemd[1]: Started Fingerprint Authentication Daemon.
```
And patchelf is getting run in `buildPhase`, is there something wrong with that or am i missing something?

Reply:
Ah! I didn't spot that. I think I'm so used to seeing the `autoPatchelfHook` that I didn't even consider it could be run manually. Feel free to mark as resolved.